examples/test_util/EclIO.cpp

```diff
@@ -0,0 +1,1179 @@
+        }
+   
+        rest=rest-num;
+       
+        if (((num < 1000) && (rest!=0)) || ((num==1000) && (rest < 0))) {
+            std::string message="Error reading binary real data, incorrect number of elements";
+            throw std::runtime_error(message);
+        }
+       
+        fileH.read((char *)&dtail, sizeof(dtail));
+        dtail=reverseInt(dtail);
+ 
+        if (dhead!=dtail){ 
+            std::string message="Error reading binary real data, tail not matching header.";
+            throw std::runtime_error(message);
+        }
+    } 
+   
+    return arr;
+}
+
+std::vector<double> EclIO::readBinaryDoubArray(std::fstream &fileH,const int ant){
+
+    std::vector<double> arr;
+    double value;
+    int num,dhead,dtail,rest;
+
+    if (!fileH.is_open()){
+        std::string message="fstream fileH not open for reading";
+        throw std::runtime_error(message);
+    }
+   
+    rest=ant;
+
+    while (rest > 0) {   
+ 
+        fileH.read((char *)&dhead, sizeof(dhead));
+        dhead=reverseInt(dhead);
+   
+        num=dhead/8;
+       
+        if ((num > 1000) || (num < 0)){
+            std::string message="Error reading binary doub data, inconsistent header data or incorrect number of elements";
+	    throw std::runtime_error(message);
+        }
+       
+        for (int i=0;i<num;i++){
+            fileH.read((char *)&value, sizeof(value));
+            value=reverseDouble(value);
+            arr.push_back(value);
+        }
+   
+        rest=rest-num;
+       
+        if (((num < 1000) && (rest!=0)) || ((num==1000) && (rest < 0))) {
+            std::string message="Error reading binary doub data, incorrect number of elements";
+            throw std::runtime_error(message);
+        }
+
+        fileH.read((char *)&dtail, sizeof(dtail));
+        dtail=reverseInt(dtail);
+ 
+        if (dhead!=dtail){ 
+            std::string message="Error reading binary doub data, tail not matching header.";
+            throw std::runtime_error(message);
+        }
+    } 
+   
+    return arr;
+}
+
+std::vector<bool> EclIO::readBinaryLogiArray(std::fstream &fileH,const int ant){
+
+    std::vector<bool> arr;
+    bool value;
+    int num,dhead,dtail,rest;
+    unsigned int intVal;
+
+    if (!fileH.is_open()){
+        std::string message="fstream fileH not open for reading";
+        throw std::runtime_error(message);
+    }
+   
+    rest=ant;
+   
+    while (rest > 0) {   
+ 
+        fileH.read((char *)&dhead, sizeof(dhead));
+        dhead=reverseInt(dhead);
+   
+        num=dhead/4;
+
+        if ((num > 1000) || (num < 0)){
+            std::string message="Error reading binary logi data, inconsistent header data or incorrect number of elements";
+	    throw std::runtime_error(message);
+        }
+
+        for (int i=0;i<num;i++){
+
+            fileH.read((char *)&intVal, 4);
+           
+            if (intVal==0xffffffff){
+                value=true;
+            } else if (intVal==0x00000000){
```

Comment:
Magic numbers.

Reply:
Are you not happy with the usage of hex representation directly in the if statement ? if true all 32 bits are 1, if false all 32 bits are 0 in the binary file.

Reply:
I want a named constant; i.e. something like:
```
int true_value = ;
int false_value = ;
```

Reply:
Added 
    const unsigned int true_value = 0xffffffff;
    const unsigned int false_value = 0x00000000;

in EclIOdata.hpp